tests/SharpCompress.Test/Tar/TarArchiveTests.cs

```diff
@@ -217,8 +217,7 @@ public void Tar_Empty_Archive()
     {
         var archiveFullPath = Path.Combine(TEST_ARCHIVES_PATH, "Tar.Empty.tar");
         using Stream stream = File.OpenRead(archiveFullPath);
-        using var archive = ArchiveFactory.Open(stream);
-        Assert.True(archive.Type == ArchiveType.Tar);
+        Assert.Throws<InvalidOperationException>(() => ArchiveFactory.Open(stream));
```

Comment:
Feels like not being able to open an empty tar is a regression

Reply:
You are correct, the empty file is a special case that needs to be taken care of, updated solution proposal